### PR TITLE
Feature/2.0.0

### PR DESCRIPTION
**Pull request by codeclout**
* 'one-var' linter rule change
* .nvmrc to 7.6+ per request
* drops configuration options for `handlebars`, `envProd` and `maxAge`
* updates example app
* fixes metrics plugin integration
* fixes isDebug flag
* adds config validation
* dependency bumps
* refactors/re-organizes the things

### CODE REVIEW

File: init.js [hunk 1]
```diff
@@ -1,162 +1,106 @@
-'use strict';
+const events = require('events');
+const Config = require('./lib/config');
+const Hapi = require('hapi');
+const Registry = require('./lib/registry');
```
Comment (adslaton) [inline init.js]: Why make this change?

Comment (codeclout) [inline init.js]: @adslaton - fixed the bug noted below... but constants placed on individual lines is just a design decision I think is neater.

**Comment by adslaton** [comment]
The issues have now been resolved.

**Comment by codeclout** [comment]
@adslaton - conflicts have been resolved

**Comment by codeclout** [comment]
merging v2.0.0

**Pull request merged**